includes/class-wc-gateway-amazon-payments-advanced.php

```diff
@@ -479,7 +479,9 @@ public function checkout_init( $checkout ) {
 		}
 
 		if ( ! $this->is_logged_in() ) {
-			add_filter( 'woocommerce_available_payment_gateways', array( $this, 'remove_amazon_gateway' ) );
+			if ( ! is_wc_endpoint_url( 'order-pay' ) ) {
```

Comment:
this two if can be one with an &&

Reply:
no, because if it's not logged in, we still need to not do the rest.  but we only remove the gateway if we're not on the order-pay screen.